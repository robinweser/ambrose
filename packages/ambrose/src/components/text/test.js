import React from 'react'

import Text from '.'

import createSnapshot from '../../testing/createSnapshot'

describe('<Text />', () => {
  it('renders correctly with default props', () => {
    const snapshot = createSnapshot(<Text>Children</Text>)
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly with variant', () => {
    const snapshot = createSnapshot(<Text variant="heading">Children</Text>)
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly with substyle', () => {
    const snapshot = createSnapshot(<Text subStyle="emphasis">Children</Text>)
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly with color', () => {
    const snapshot = createSnapshot(<Text color="green">Children</Text>)
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly when nested', () => {
    const snapshot = createSnapshot(
      <Text>
        Nested <Text subStyle="emphasis">Children</Text>
      </Text>
    )
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly when aligned', () => {
    const snapshot = createSnapshot(<Text align="right">Children</Text>)
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly when transformed', () => {
    const snapshot = createSnapshot(
      <Text>
        Nested <Text transform="uppercase">children</Text>
      </Text>
    )
    expect(snapshot).toMatchSnapshot()
  })
  it('renders correctly when nesting colosr', () => {
    const snapshot = createSnapshot(
      <Text color="primary">
        Nested <Text subStyle="emphasis">Children</Text>
      </Text>
    )
    expect(snapshot).toMatchSnapshot()
  })
})
