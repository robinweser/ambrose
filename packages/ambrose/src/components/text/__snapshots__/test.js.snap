// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders correctly when aligned 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  font-size: 12px;
  margin: 0;
  display: block;
  color: black;
  text-align: right;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;

exports[`<Text /> renders correctly when nested 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  font-size: 12px;
  margin: 0;
  display: block;
  color: black;
}
.cls_2 {
  font-size: 12px;
  font-weight: 700;
  margin: 0;
  color: black;
}


<div className=test-theme cls_0>
  <div className=cls_1>
    Nested <span className=cls_2>Children</span>
  </div>
</div>;
"
`;

exports[`<Text /> renders correctly when nesting colosr 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  font-size: 12px;
  margin: 0;
  display: block;
  color: primary;
}
.cls_2 {
  font-size: 12px;
  font-weight: 700;
  margin: 0;
  color: primary;
}


<div className=test-theme cls_0>
  <div className=cls_1>
    Nested <span className=cls_2>Children</span>
  </div>
</div>;
"
`;

exports[`<Text /> renders correctly when transformed 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  font-size: 12px;
  margin: 0;
  display: block;
  color: black;
}
.cls_2 {
  font-size: 12px;
  margin: 0;
  color: black;
  text-transform: uppercase;
}


<div className=test-theme cls_0>
  <div className=cls_1>
    Nested <span className=cls_2>children</span>
  </div>
</div>;
"
`;

exports[`<Text /> renders correctly with color 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  font-size: 12px;
  margin: 0;
  display: block;
  color: green;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;

exports[`<Text /> renders correctly with default props 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  font-size: 12px;
  margin: 0;
  display: block;
  color: black;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;

exports[`<Text /> renders correctly with substyle 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  font-size: 12px;
  font-weight: 700;
  margin: 0;
  display: block;
  color: black;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;

exports[`<Text /> renders correctly with variant 1`] = `
".cls_0 {
  display: contents;
}
.cls_1 {
  font-size: 20px;
  margin: 0;
  display: block;
  color: black;
}


<div className=test-theme cls_0>
  <div className=cls_1>Children</div>
</div>;
"
`;
